Source/VCGeneration/Split.cs

```diff
@@ -1071,6 +1066,159 @@ bool ContainsAssert(Block b)
         return splits;
       }
 
+      public static List<Split> FocusImpl(Implementation impl, Dictionary<TransferCmd, ReturnCmd> gotoCmdOrigins, VCGen par)
+      {
+        bool isFocusCmd(Cmd c) {
+          return c is PredicateCmd p && QKeyValue.FindBoolAttribute(p.Attributes, "focus");
+        }
+        List<Block> getFocusBlocks(List<Block> blocks) {
+          return blocks.Where(blk => blk.Cmds.Where(c => isFocusCmd(c)).Count() != 0).ToList();
+        }
+        var dag = Program.GraphFromImpl(impl);
+        var topoSorted = dag.TopologicalSort().ToList();
+        // If reallyFocus is set to true,
+        // foci are processed in a top-down fashion --- i.e., if two foci are on the same path,
+        // the ancestor is processed first.
+        // On the other hand, if reallyFocus is false,
+        // foci are processed in a bottom-up fashion --- i.e., the descendant is processed first.
+        bool reallyFocus = true;
```

Comment:
Any particular reason why you chose not to make "reallyFocus" specifiable by a Boogie user, potentially by making it a command-line parameter?  Perhaps you just want to experiment for a while before really deciding whether this option is worth it.